### PR TITLE
Refactor topic handling in log()

### PR DESCRIPTION
**Pull request by axic**
Part of #156.

Inspired by #148.

Helps with #294.

### CODE REVIEW

File: src/eei.cpp [hunk 1]
```diff
@@ -481,18 +481,23 @@ inline int64_t maxCallGas(int64_t gas) {
...
+      uint32_t topic3 = arguments[5].geti32();
+      uint32_t topic4 = arguments[6].geti32();
 
       HERA_DEBUG << "log " << hex << dataOffset << " " << length << " " << numberOfTopics << dec << "\n";
 
       ensureCondition(!(msg.flags & EVMC_STATIC), StaticModeViolation, "log");
 
       ensureCondition(numberOfTopics <= 4, ContractValidationFailure, "Too many topics specified");
 
+      // FIXME: should this assert that unused topic offsets must be 0?
```
Comment (axic) [inline src/eei.cpp]: Created an issue: https://github.com/ewasm/design/issues/129

**Comment by chfast** [comment]
I'm not sure where it is going.

**Comment by axic** [comment]
> I'm not sure where it is going.

See #294.

These will be moved into their individual methods. Then `EthereumInterface` will be a class of its own without any binaryen influence and all the VMs will have their own subclass inheriting from it, implementing the `memory*` helpers (will be virtual then) and mapping the `eii*` functions in their respective host function wrappers.

**Comment by chfast** [comment]
Ok, so `topicX` are going to be passed to a function, not abstracting only memory access.

**Comment by axic** [comment]
Correct, the host function wrappers need to acquire the arguments as i32/i64 pass down to the actual implementation and return i32/i6.

**Pull request merged**